File: LeetCode-DSA/AugustChallenge_2022/PalindromeLinkedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) 
    {
        vector<int> orig;
        while(head != NULL)
        {
            orig.push_back(head -> val);
            head = head -> next;
        }
        int start = 0;
        int end = orig.size() - 1;
        while(start <= end)
        {
            if(orig[start] == orig[end])
            {
                start++;
                end--;
            }
            else
                return false;
        }
        return true;
    }
};

File: LeetCode-DSA/LeetCode75/Level2/PalindromeLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

'''
Q234. Description: Given the head of a singly linked list, return true if it is a palindrome or false otherwise.
'''

class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        # Convert linked list to string and compare with reverse
        data = ''
        while(head):
            data += (str)(head.val)
            head = head.next
        return data == data[::-1]